kotlinx-coroutines-core/jvm/src/CoroutineContext.kt

```diff
@@ -217,15 +217,18 @@ internal actual class UndispatchedCoroutine<in T>actual constructor (
     }
 
     fun clearThreadContext(): Boolean {
-        if (threadStateToRecover.get() == null) return false
-        threadStateToRecover.set(null)
+        if (threadStateToRecover.get() == null) {
+            threadStateToRecover.remove()
+            return false
+        }
+        threadStateToRecover.remove()
         return true
     }
 
     override fun afterResume(state: Any?) {
         threadStateToRecover.get()?.let { (ctx, value) ->
```

Comment:
isn't it needed to call `remove()` even if `get()` returns `null` (by analogy with `clearThreadContext`?

Reply:
Nice catch!